gen/version.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+//go:generate go run version.go
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"text/template"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/storer"
+	"golang.org/x/mod/modfile"
+)
+
+const versionFileTemplate = `// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package configuration
+
+const (
+	RosettaVersion    = "{{ .RosettaVersion }}"
+	NodeVersion       = "{{ .NodeVersion }}"
+	MiddlewareVersion = "{{ .MiddlewareVersion }}"
```

Comment:
Correct me if I'm wrong, but wouldn't this leave the space in front and behind the variable? IMO, you need to use `{{-` and `-}}` to trim space before/after the template variable?

As far as I know, Go templates don't use the space in-between, i.e. `{{.RosettaVersion}}`.

Reply:
It works just fine with spaces, I find it more readable but if you guys prefer it without the spaces I'm not against it.

Here is an [example showing that the dashes are not necessary](https://goplay.space/#-rW641WpAvq), and as you can see for example [in the go source code itself](https://golang.org/src/text/template/example_test.go), the dashes are only needed to prevent line breaks when using code statement (`if`, `else`, `with`, `end`, etc.). When we're just inserting a value in a line, it is not needed.

Reply:
I don't mind then, but just FYI, none of the reference examples in the standard library use spaces in-between.

Reply:
@Maelkum Any opinion on this, since we're both fine either way, you can take the final decision! :p

Reply:
I find the version with the spaces a bit more readable too, if it works just the same.